flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperator.java

```diff
@@ -335,7 +335,11 @@ public void open() throws Exception {}
      * @throws Exception An exception in this method causes the operator to fail.
      */
     @Override
-    public void close() throws Exception {}
+    public void close() throws Exception {
+        if (output != null) {
+            output.close();
+        }
+    }
```

Comment:
I think the `OperatorChain` should be the owner of the outputs. He is referencing some of the created outputs and using them for other things:
- registering some metrics (maybe can be safely ignored?)
- `OperatorChain#chainedSources` are also referencing some outputs

Hence I would suggest that `OperatorChain#releaseOutputs()` should be the guy to close all outputs that were created inside that `OperatorChain`.

Reply:
Yes, I agree.